src/DashboardService.Data/TaskTypeRepository.cs

```diff
@@ -18,28 +21,83 @@ public TaskTypeRepository(IDataProvider provider)
     _provider = provider;
   }
   
-  public Task<Guid?> CreateAsync(DbTaskType dbDepartment)
+  public async Task<Guid?> CreateAsync(DbTaskType dbTaskType)
   {
-    throw new NotImplementedException();
+    _provider.TaskTypes.Add(dbTaskType);
+    await _provider.SaveAsync();
+
+    return dbTaskType.Id;
   }
 
-  public Task<List<DbTaskType>> GetAllAsync(GetTaskTypesFilter filter)
+  public async Task<(List<DbTaskType> dbTaskTypes, int totalCount)> GetAllAsync(GetTaskTypesFilter filter, CancellationToken ct)
   {
-    throw new NotImplementedException();
+    IQueryable<DbTaskType> taskTypes = _provider.TaskTypes.AsNoTracking();
+
+    if (filter.IsAscendingSort.HasValue)
+    {
+      taskTypes = filter.IsAscendingSort.Value
+        ? taskTypes.OrderBy(d => d.Name)
+        : taskTypes.OrderByDescending(d => d.Name);
+    }
+
+    if (!string.IsNullOrWhiteSpace(filter.NameIncludeSubstring))
+    {
+      taskTypes = taskTypes.Where(d => d.Name.Contains(filter.NameIncludeSubstring));
+    }
+
+    return (
+      await taskTypes
+        .Skip(filter.SkipCount)
+        .Take(filter.TakeCount)
+        .ToListAsync(ct),
+      await taskTypes.CountAsync(ct));
   }
 
-  public Task<DbTaskType> GetAsync(Guid id)
+  public async Task<DbTaskType> GetAsync(Guid id, CancellationToken ct = default)
   {
-    throw new NotImplementedException();
+    return await _provider.TaskTypes.AsNoTracking().SingleOrDefaultAsync(tt=>tt.Id == id, ct);
+  }
+
+  public async Task<bool> EditAsync(Guid id, JsonPatchDocument<DbTaskType> request, CancellationToken ct)
+  {
+    DbTaskType dbTaskType = await GetAsync(id, ct);
+
+    if (dbTaskType is null)
+    {
+      return false;
+    }
+
+    request.ApplyTo(dbTaskType);
+
+    await _provider.SaveAsync();
+
+    return true;
+  }
+
+  public async Task<bool> RemoveAsync(Guid id, CancellationToken ct)
+  {
+    DbTaskType dbTaskType = await GetAsync(id, ct);
+
+    if (dbTaskType is null)
+    {
+      return false;
+    }
+
+    _provider.TaskTypes.Remove(dbTaskType);
+    await _provider.SaveAsync();
```

Comment:
ct?

Reply:
for SaveAsync no overloads with ct as argument

Reply:
we dont have in IBaseDataProvider overloads for method Task SaveAsync() with ct as argument and i cant use here ct. only if add same method to IDataProvider, but i dont think that this is correct class for such method

Reply:
Yes, I mixed it up with the method SaveChangesAsync of DbContext